src/components/WrapEtherBtn.tsx

```diff
@@ -0,0 +1,280 @@
+      tooltipText: 'Unwrap converts back WETH into ETH',
+    }
+  }
+}
+
+const WrapUnwrapEtherBtn: React.FC<WrapUnwrapEtherBtnProps> = (props: WrapUnwrapEtherBtnProps) => {
+  const { wrap, label, className } = props
+  const [wethHelpVisible, showWethHelp] = useSafeState(false)
+  const { register, errors, setValue, watch, triggerValidation } = useForm()
+  // const formRef = useRef<HTMLFormElement | null>(null)
+  const amountValue = watch(INPUT_ID_AMOUNT)
+  const amountError = errors[INPUT_ID_AMOUNT]
+
+  // console.log('amountValue', amountValue)
+
+  const { title, balance, symbolSource, tooltipText, description, amountLabel } = useMemo(
+    () => getModalParams(wrap, wethHelpVisible, showWethHelp),
+    [wrap, wethHelpVisible, showWethHelp],
+  )
+
+  // const onSubmit = (data: any, event?: BaseSyntheticEvent): void => {
+  //   // OnSubmit<FieldValues>
+  //   alert('on submit 1')
+  //   if (event) {
+  //     alert('on submit 2')
+  //     console.log('Stop propagation')
+  //     event.preventDefault()
+  //     event.stopPropagation()
+  //   }
+  //   alert('on submit 4')
+  //   console.log('HANDLE SUBMITTT!!!', data)
+  // }
+
+  // const handleSubmit = (event: BaseSyntheticEvent): void => {
+  //   alert('A name was submitted')
+  //   event.preventDefault()
+  // }
+
+  const [modalHook, toggleModal] = useModali({
+    ...DEFAULT_MODAL_OPTIONS,
+    title,
+    message: (
+      <ModalWrapper>
+        <div>
+          {description}
+          <b>Available {symbolSource}</b>
+          <div>
+            <a onClick={(): void => setValue(INPUT_ID_AMOUNT, formatAmountFull(balance), true)}>
+              {formatAmountFull({ amount: balance, precision: DEFAULT_PRECISION }) || ''} {symbolSource}
+            </a>
+          </div>
+        </div>
+        <div>
+          <b>{amountLabel}</b>
+          <div>
+            <InputBox>
+              <i>{symbolSource}</i>
+              <input
+                type="text"
+                name={INPUT_ID_AMOUNT}
+                placeholder="0"
+                required
+                ref={register({
+                  pattern: { value: validInputPattern, message: 'Invalid amount' },
+                  validate: { positive: validatePositiveConstructor('Invalid amount') },
+                  required: 'The amount is required',
+                  min: 0,
+                })}
+              />
+            </InputBox>
+          </div>
+          {amountError && <p className="error">Invalid amount</p>}
+        </div>
+        {/* <form ref={formRef} onSubmit={handleSubmit}></form> */}
+      </ModalWrapper>
+    ),
+    buttons: [
+      <Modali.Button label="Cancel" key="no" isStyleCancel onClick={(): void => toggleModal()} />,
+      <Modali.Button
+        label="Continue"
+        key="yes"
+        isStyleDefault
+        onClick={(): void => {
+          // if (formRef.current) {
+          //   formRef.current.submit()
+          // }
+          // handleSubmit()
```

Comment:
You should be able to do `handleSubmit(onSubmit())()`
https://react-hook-form.com/api/#handleSubmit

Reply:
`handleSubmit(onSubmit())()()()()()` :laughing: :rofl: :upside_down_face: :laughing:

Reply:
@W3stside  did exactly that :+1: